bc_obps/common/models/base_model.py

```diff
@@ -24,3 +26,26 @@ def save(self, *args, **kwargs) -> None:
         else:
             self.full_clean(exclude=AUDIT_FIELDS)  # validate the model before saving
         super().save(*args, **kwargs)
+
+    @typing.no_type_check
+    def custom_update_or_create(self, user_guid: UUID, **kwargs: dict):
+        # Extract pk or id from kwargs if available
+        pk = kwargs.pop('pk', None)
+        kwargs_id = kwargs.pop('id', None)
```

Comment:
@BCerki  was there any particular reason why you chose to use `pop` here instead of `kwargs.get()`? pop() removed the key-value from the dictionary, which seems unnecessary

Reply:
This is @Sepehr-Sobhani 's code, so I'm not sure, sorry! (Brianna isn't actually a python expert :see_no_evil:)

Reply:
all good, i re-read the code and now understand why. It's because a few lines lower where we're updating or creating a record with the data stored in `kwargs`, we don't want to overwrite the existing pk or id if one already exists.

P.S. you're gonna have to try harder than that to shake your title of Python Expert